src/pages/index.js

```diff
@@ -24,40 +36,166 @@ const profileEditModalClass = new ModalWithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit
 );
+const profileEditAvatarModalClass = new ModalWithForm(
+  "#profile-edit-avatar-modal",
+  handleProfileAvatar
+);
 
+const confirmationModalClass = new ModalWithConfirmation("#confirmation-modal");
 const previewImageModalClass = new ModalWithImage("#preview_image_modal");
 
-// Functions
+// GET API
 
-const sectionClass = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
-sectionClass.renderItems();
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfoClass.setUserInfo({ title: data.name, description: data.about });
+    userInfoClass.setUserAvatar({
+      avatar: data.avatar,
+    });
 
-function handleProfileFormSubmit(formData) {
-  userInfoClass.setUserInfo(formData);
-}
+    return data;
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
 
-function handleImageClick(cardData) {
-  previewImageModalClass.open(cardData);
-}
+let sectionClass;
+
+api
+  .getInitialCards()
+  .then((cardList) => {
+    sectionClass = new Section(
+      { items: cardList, renderer: createCard },
+      ".cards__list"
+    );
+    sectionClass.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
 
-function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleImageClick);
-  return cardElement.getCardElement();
+// PATCH API
+function handleProfileFormSubmit(formData) {
+  profileEditModalClass.setLoading(true, "Save");
+  api
+    .updateProfileInfo(formData.title, formData.description)
+    .then((data) => {
+      userInfoClass.setUserInfo({ title: data.name, description: data.about });
+      profileEditModalClass.close();
+
+      return data;
+    })
+    .catch((error) => {
+      console.error("Error fetching user info:", error);
+    })
+    .finally(() => {
+      profileEditModalClass.setLoading(false, "Save");
+      profileEditForm.reset();
+      formValidators["modal-edit-form"].disableButton();

```

Comment:
* You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.
* You need to disable the button only in block `then`, otherwise you'll not be able to click it if there is an error